### PR TITLE
use tools.deps.alpha map

### PR DESCRIPTION
**Pull request by mikepjb**
I tested this by creating a new folder with a blank deps.edn then using:

`clojure -Sdeps '{:deps {pack/pack.alpha {:git/url "https://github.com/juxt/pack.alpha.git" :sha "dccf2134bcf03726a9465d2b9997c42e5cd91bff"}}}' -m mach.pack.alpha.inject '2769a6224bfb938e777906ea311b3daf7d2220f5'`

Which injected pack.alpha into the deps.edn (I then updated this to `:local/root` to point at my local copy of the project

`clj -A:pack mach.pack.alpha.capsule ok.jar`

and running:
`java -jar ok.jar`
which presented me with a Clojure REPL

There was a complain about missing StaticLoggerBinder so a little more work is needed - here is the error output:
```
test(!) $ clj -A:pack mach.pack.alpha.capsule ok.jar
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
test(!) $ 
```

### CODE REVIEW

**Comment by SevereOverfl0w** [comment]
I don't think that message is new :)

**Comment by SevereOverfl0w** [comment]
Oh, yes it is. tools.deps.alpha removed their dependency on slf4j-nop. We're now in the awkward position of having to either bring along our own slf4j solution (e.g. -nop) or none at all and leave it to the user.

Technically speaking, the message that's there is OK, if unattractive.

**Comment by mikepjb** [comment]
I've included slf4j-nop (basically the reverse of Alex's 00dfa4a113311555ea9738889a295c88213253b3 on tools.deps.alpha)

This removes the warning (still with a successfully build uberjar)

**Comment by mikepjb** [comment]
I suppose a consequence of including nop is that anyone wanting to use slf4j will need to add nop as an exclusion in their deps.edn file?

**Comment by SevereOverfl0w** [comment]
Only if the message bothers them. There's a potential risk that they will end up seeing lots of output otherwise, depending on their config.

This isn't something I have a great solution for unfortunately. There's a related open issue.

File: src/mach/pack/alpha/impl/tools_deps.clj [hunk 1]
```diff
@@ -55,10 +55,7 @@
 
 (defn system-edn
   []
-  (-> "mach/pack/alpha/system_deps.edn"
-      io/resource
-      slurp
-      edn/read-string))
+  (tools.deps.reader/read-deps (tools.deps.reader/default-deps)))
```
Comment (SevereOverfl0w) [inline src/mach/pack/alpha/impl/tools_deps.clj]: I don't want to use the default-deps here, I don't want `:deps` from the user's `~/.clojure/deps.edn` to be factored into creating uberjars. This should be the install-deps instead.

Comment (mikepjb) [reply src/mach/pack/alpha/impl/tools_deps.clj]: I've changed this argument to nil as read-deps merges it's arguments into (install-deps) anyway.

Comment (mikepjb) [reply src/mach/pack/alpha/impl/tools_deps.clj]: This has been updated again (system-edn is no longer needed as install-deps can be inlined)

**Comment by SevereOverfl0w** [comment]
Sorry, dropped the ball on this.

**Pull request merged**